common/unsafe/src/main/java/org/apache/spark/unsafe/memory/OnHeapMemoryBlock.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.unsafe.memory;
+
+import org.apache.spark.unsafe.Platform;
+
+/**
+ * A consecutive block of memory with a long array on Java heap.
+ */
+public final class OnHeapMemoryBlock extends MemoryBlock {
```

Comment:
I think this class could be named as ```LongArrayMemoryBlock```, in order to correspond to ```ByteArrayMemoryBlock```, as they are both on-heap memory block.

Reply:
Originally, the name was `LongArrayMemoryBlock`. After the discussion with @cloud-fan, we decided to named as `OnHeapMemoryBlock` compared to `OffHeapMemoryBlock`.

WDTY?

Reply:
I still suggest ```LongArrayMemoryBlock```.

Reply:
Thank you for your suggestion. I would like to hear other opinions, too.

Reply:
It's okay.

Reply:
So the basic idea is, `OnHeapMemoryBlock` and `OffHeapMemoryBlock` are the major memory blocks that should be used inside Spark, returned by the memory manager according to the memory mode. `ByteArrayMemoryBlock` would only be used in some certain places where data is always `bye[]`, and the memory is not allocated from the memory manager. Hopefully we can always use memory returned by memory manager and track the memory usage, and eventually remove `ByteArrayMemoryBlock`.

Reply:
👍

Reply:
Can we put @cloud-fan's comment https://github.com/apache/spark/pull/19222/files#r172356146 into `OnHeapMemoryBlock`'s doc?